pkg/checkpoint/checkpoint_restore.go

```diff
@@ -95,6 +96,23 @@ func CRImportCheckpoint(ctx context.Context, runtime *libpod.Runtime, restoreOpt
 		newName = true
 	}
 
+	if len(restoreOptions.PublishPorts) > 0 {
+		ctrConfig.PortMappings = []ocicni.PortMapping{}
+		for _, pp := range restoreOptions.PublishPorts {
+			mapping := ocicni.PortMapping{
```

Comment:
This actually needs to be a loop - a `specgen.PortMapping` represents a range of ports (the `Range` field), and we need to generate one `ocicni.PortMapping` for each.

Reply:
You may want to use https://github.com/containers/podman/blob/master/pkg/specgen/generate/ports.go#L27 (it's in a problematic package right now - would introduce an import loop - but I don't see why it can't be moved to `/pkg/specgen` and that ought to be OK.

Reply:
Thanks for the pointer to existing code. I was grepping a lot but I was not able to find it.

About 'This actually needs to be a loop'. It is a loop. At least I hope it is. Line 101 has `for _, pp := range restoreOptions.PublishPorts {` That should loop over all port mappings, right?

Reply:
My original thought with the "needs to be a loop" comment was you needed a nested loop - loop over `PublishedPorts` and then loop over each port's `Range` to generate a number of ocicni PortMapping structs for each port range in PublishedPorts. The function I linked should remove that need and do the work itself.

Reply:
Switched to the existing function. Thanks for the pointer.